nlprule/src/tokenizer.rs

```diff
@@ -245,75 +290,79 @@ impl Tokenizer {
 
     /// Tokenize the given sentence. This applies chunking and tagging, but does not do disambiguation.
     // NB: this is not public because it could be easily misused by passing a text instead of one sentence.
-    pub(crate) fn tokenize<'t>(&'t self, sentence: &'t str) -> Vec<IncompleteToken<'t>> {
-        let mut current_char = 0;
+    pub(crate) fn tokenize<'t>(&'t self, sentence: &'t str) -> Option<IncompleteSentence<'t>> {
+        if sentence.trim().is_empty() {
+            return None;
+        }
+
         let token_strs = self.get_token_strs(sentence);
+
         let mut tokens: Vec<_> = token_strs
             .iter()
             .enumerate()
-            .map(|(i, x)| {
-                let char_start = current_char;
-                let ptr = x.as_ptr() as usize;
-                current_char += x.chars().count();
+            .filter(|(_, token_text)| !token_text.trim().is_empty())
+            .map(|(i, token_text)| {
+                let byte_start = token_text.as_ptr() as usize - sentence.as_ptr() as usize;
```

Comment:
```suggestion
                let byte_start = (token_text.as_ptr() as usize).saturating_sub(sentence.as_ptr() as usize);
```

Reply:
I agree that this is problematic but I think a `debug_assert` which checks the expectation and a comment linking to https://stackoverflow.com/questions/38268529/how-to-get-the-byte-offset-between-str is a better fix since it is strictly a bug if `sentence.as_ptr()` is larger than `token_text.as_ptr()`, I don't want to silently ignore that.

Reply:
I'd recommend going for `checked_sub(..)` then and verify the result is `None/Some(_)`, production will give you the uggliest input you can imagine, that you will never encounter in test cases.

Reply:
`checked_sub` is neat! I didn't know about that. I've changed that, changed `get_token_strs` to `get_token_ranges` and moved the pointer arithmetic there to make it more self-contained.